site/fluxyaml-config-files.md

```diff
@@ -0,0 +1,234 @@
+2. Update the manifest files of [workload](https://github.com/weaveworks/flux/blob/master/site/fluxctl.md#what-is-a-workload).
+   Specifically, flux can update:
+    * container images, when releasing a new image version. A release can happen manually or automatically, when a new
+      container image is pushed to a repository.
+    * annotations, which establish the release policy of a workload (e.g. whether it should be automatically released,
+      whether it should be locked from releasing, what image tags should be considered for automated releases …)
+
+Flux can be configured to confine the scope of (1) and (2):
+
+*   To specific (sub)directories (flag` --git-path`)
+*   To a Git branch other than `master` (flag` --git-branch`)
+
+
+## Abstracting out file-access: generators and updaters
+
+Flux allows you to declare configuration files to override file operations (1) and (2), by declaring commands which
+perform equivalent actions.
+
+The configuration files are formatted in `YAML` and named `.flux.yaml`. They must be located on the Git repository
+(more on this later).
+
+A `commandUpdated` `.flux.yaml` file has the following loosely specified format:
+
+
+```
+version: X
+commandUpdated:
+  generators:
+    - command: generator_command1 g1arg1 g1arg2 ...
+    - command: generator_command2 g2arg2 g2arg2 ...
+  updaters:
+    - containerImage:
+        command: containerImage_updater_command1 ciu1arg1 ciu1arg2 ...
+      policy:
+        command: policy_updater_command1 pu1arg1 pu1arg2 ...
+    - containerImage:
+        command: containerImage_updater_command2 ciu2arg1 ciu2arg2 ...
+      policy:
+        command: policy_updater_command2 pu2arg1 pu2arg2 ...
+```
+
+
+> **Note:** For a simpler approach to updates, Flux provides a `patchUpdated` configuration file variant.
+
+
+The file above is versioned (in order to account for future file format changes). Current version is `1`,
+which is enforced.
+
+Also, the file contains two generators (declared in the `generators `entry), used to generate manifests and two updaters
+(declared in the `updaters `entry), used to update resources in the Git repository.
+
+The generators are meant as an alternative to Flux manifest reads (1). Each updater is split into a `containerImage`
+command and a `policy` command, covering the corresponding two types of workload manifest updates mentioned in (2).
+
+> **Note** Update commands operate on policies, rather than annotations. That is for two reasons:
+>
+> * It is an implementation detail for Kubernetes manifests specifically that policies are represented as annotations.
+> * Some configurations (even those for Kubernetes clusters) may encode policies symbolically.
+
+Here is a specific `.flux.yaml` example, declaring a generator and an updater using [Kustomize](https://github.com/kubernetes-sigs/kustomize)
+(see [https://github.com/weaveworks/flux-kustomize-example](https://github.com/weaveworks/flux-kustomize-example)
```

Comment:
https://github.com/weaveworks/flux-kustomize-example doesn't exist yet, I am waiting for someone with admin rights to create it ( I want to transfer https://github.com/2opremio/flux-kustomize-demo there )